runtime/vm/UpcallVMHelpers.cpp

```diff
@@ -86,7 +102,14 @@ native2InterpJavaUpcallJ(J9UpcallMetaData *data, void *argsListPointer)
 float JNICALL
 native2InterpJavaUpcallF(J9UpcallMetaData *data, void *argsListPointer)
 {
-	return (float)0;
+	J9FloatPatternInfo floatPatternInfo;
+	UDATA returnValue = native2InterpJavaUpcallImpl(data, argsListPointer);
+	/* The value returned from the upcall method is literally the single precision (32bit) IEEE 754 floating-point
+	 * representation which must be converted to a real float value before returning back to the native
+	 * function in the downcall.
+	 */
+	floatPatternInfo.intValue = (I_32)returnValue;
```

Comment:
this could simply be a reinterpret cast, same for the rest

Reply:
Agreed and replaced with `reinterpret_cast` in `native2InterpJavaUpcallF` and `native2InterpJavaUpcallD`.